### PR TITLE
Tgui 4.3

### PR DESCRIPTION
**Pull request by ZeWaka**
[tooling] [bug] [feat]

## About the PR 

tgui 4.3 + hotfix + blink
see:
https://github.com/tgstation/tgstation/pull/56080 and https://github.com/tgstation/tgstation/pull/56225

## Why's this needed? 

keep up to date
tested locally, all works

### CODE REVIEW

File: tgui/packages/tgui/components/Blink.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { Component } from 'inferno';
+
+const DEFAULT_BLINKING_INTERVAL = 1000;
+const DEFAULT_BLINKING_TIME = 1000;
+
+export class Blink extends Component {
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/components/Blink.js:6]: Thanks, I hate it.

File: tgui/packages/tgui/debug/KitchenSink.js [hunk 2]
```diff
@@ -4,94 +4,28 @@
...
-];
+/**
+ * @returns {{
+ *   meta: {
+ *     title: string,
+ *     render: () => any,
+ *   },
+ * }[]}
+ */
+const getStories = () => r.keys().map(path => r(path));
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/debug/KitchenSink.js:21]: Use of story files good. 👍

File: tgui/packages/tgui/styles/components/Section.scss [hunk 3]
```diff
@@ -57,26 +61,39 @@ $shadow-offset: 0 0 !default;
...
 
 .Section--scrollable {
   overflow-x: hidden;
+  overflow-y: hidden;
+}
+
+.Section--scrollable .Section__content {
   overflow-y: scroll;
+  overflow-x: hidden;
 }
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Section.scss:97]: Not taking advantage of SCSS nesting here?
```suggestion
.Section--scrollable {
  overflow-x: hidden;
  overflow-y: hidden;

  .Section__content {
    overflow-y: scroll;
    overflow-x: hidden;
  }
}
```

File: tgui/bin/tgui-dev-server.bat [hunk 4]
```diff
@@ -1,7 +1,7 @@
 @echo off
 rem Copyright (c) 2020 Aleksej Komarov
 rem SPDX-License-Identifier: MIT
-call powershell.exe -NoLogo -ExecutionPolicy Bypass -File "%~dp0\tgui.ps1" --dev %*
+call powershell.exe -NoLogo -ExecutionPolicy Bypass -File "%~dp0\tgui_.ps1" --dev %*
```
[the remark this answers is not in the document]
Comment (ZeWaka) [reply tgui/bin/tgui-dev-server.bat:4]: so it doesn't conflict with the .bat apparently?

**Review by mordent-goonstation — APPROVED** [review]

**Comment by mordent-goonstation** [comment]
Caveat: I didn't read through all of the component stories, assumed they were mostly just cut+paste from the kitchen sink.

File: tgui/bin/tgui [hunk 5]
```diff
@@ -140,16 +145,6 @@ if [[ ${1} == "--install-git-hooks" ]]; then
...
-## Continuous integration scenario
-if [[ ${1} == "--ci" ]]; then
-  task-clean
-  task-install
-  task-eslint
-  task-webpack --mode=production
-  task-validate-build
-  exit 0
-fi
-
```
Comment (pali6) [inline tgui/bin/tgui:152]: This along with the change to build_tgui.sh seems to have disabled the github CI check for unbuilt bundle.

**Pull request merged**